In [5]:
import math
def numAnl_11_1a(x):
    return (x*math.cos(x))-(2*x**2)+(3*x)-1

print(numAnl_11_1a(0.2), numAnl_11_1a(0.3))
print(numAnl_11_1a(1.2), numAnl_11_1a(1.3))

-0.28398668443175157 0.0066009467376817454
0.154829305372008 -0.13225152278803654


In [7]:
def numAnl_11_7a(x):
    return 1-math.e**x+(math.e-1)*math.sin((math.pi/2)*x)

print(numAnl_11_7a(0), numAnl_11_7a(1))

0.0 0.0


In [8]:
def numAnl_11_7a_der(x):
    return -math.e**x+(math.pi/2)*(math.e-1)*math.cos((math.pi/2)*x)

print(numAnl_11_7a_der(0), numAnl_11_7a_der(1))

1.6990707845418864 -2.718281828459045


# Numerical Analysis: Burden, Faires
## Chapter 2

### Examples
#### 1
Show that $f(x)=x^3+4x^2-10=0$ has a root in $[1,2]$ and use the Bisection method to determine an approximation to the root that is accurate to at least within $10^{-4}$

**Solution**:

Because $f(1)=-5$ and $f(2)=14$, the Intermediate Value Theorem ensure that this continuous function has a root in $[1,2]$.

For the first iteration of the Bisection method, we use the fact that at the midpoint of $[1,2]$, we have $f(1.5)=2.375 > 0$. This indicates we should select the interval $[1,1.5]$ for our second iteration. Then we find that $f(1.25)=-1.796875 < 0$. This indicates we should select the interval $[1.25,1.5]$ whose midpoint is $1.375$. Continuing in this manner gives us the following table.



In [8]:
def numAnl_21_3(a,b):
    x=(a+b)/2
    return x**3-7*x**2+14*x-6

# print(numAnl_21_3(3.2), numAnl_21_3(4))
numAnl_21_3(3.4,3.425)

-0.0019980468750020464

In [9]:
(3.4+3.425)/2

3.4124999999999996

### Exercise 2.1
#### 3(c)
$f(x)=x^3-7x^2+14x-6=0$
Interval: $[3.2, 4]$

$f(3.2)=-0.11$ and $f(4)=2$, the Intermediate Value Theorem ensure that this continuous function has a root in $[3.2,4]$.

The following table is generated for the bisection method:

|  $n$  |  $a_n$  |  $b_n$  |  $p_n$   |  $f(p_n)$  |
|-------|---------|---------|----------|------------|
| $1$   | $3.2$   | $4$     | $3.6$    | $0.336$    |
| $2$   | $3.2$   | $3.6$   | $3.4$    | $-0.016$   |
| $3$   | $3.4$   | $3.6$   | $3.5$    | $0.125$    |
| $4$   | $3.4$   | $3.5$   | $3.45$   | $0.046$    |
| $5$   | $3.4$   | $3.45$  | $3.425$  | $0.013$    |
| $6$   | $3.4$   | $3.425$ | $3.4125$ | $-0.002$   |

The Solution is 3.4125.

### Exercise 2.2
#### 7


## 1. Modeling, Computers, and Error Analysis
1.1.5 Introduction to Error Analysis
Sources of Error
- Modeling Error
- Numerical Error
    - Truncation Error
    - Roundoff Error

Accuracy Vs. Precision:
- Accuracy is how close to the bulls eye you hit.
- Precision is how narrowed or spreaded out your hits are.

True Error = True value - Approximation

Relateive error = (True Error / True Value) * 100

Approximate Error = (Current Approximation - Previous Approximation) / Approximate Value


### Mathematical Modelling and Engineering Problem Solving
### Programming and Software
### Approximations and Round-off Errors
### Truncation Errors and the Taylor Series

## 2. Roots of Equations
### Bracketing Methods
### Open Methods
### Roots of Polynomials
### Case Studies: Roots of Equations

## 3. Linear Algebraic Equations
### Gauss Elimination
### LU Decomposition and Matrix Inversion
### Special Matrices and Gauss-Sidel
### Case Studies: Linear Algebraic Equations
    
## 4. Optimization
### 1-D Unconstrained Optimization
### Multidimensional Unconstrained Optimization
### Constrained Optimization
### Case Studies: Optimization

## 5. Curve Fittinn
### Least-Squares Regression
### Interpolation
### Fourier Approximation
### Case Studies: Curve Fitting
    
## 6. Numeric Differentiation and Integration
### Newton-Cotes Integration Formulas
### Integration of Equations
### Numerical Differentiation
### Case Studies: Numerical Integration and Differentiation
    
## 7. Odrinary Differential Equations
### Runge-Kutta Methods
### Stiffness and Multistep Methods
### Boundary-Value and Eigenvalue Problems
### Case Studies: Ordinary Differential Equations
    
## 8. Partial Differential Equations
### Finite Difference: Elliptic Equations
### Finite Difference: Parabolic Equations
### Finite-Eliment Method
### Case Studies: 
